smbprotocol/file_info.py

```diff
@@ -877,3 +877,23 @@ def __init__(self):
             )),
         ])
         super(FileFsVolumeInformation, self).__init__()
+
+
+class FileFsFullSizeInformation(Structure):
+    """
+    [MS-FSCC] 2.5.4 FileFsFullSizeInformation
+    https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/63768db7-9012-4209-8cca-00781e7322f5
+    """
+
+    INFO_TYPE = InfoType.SMB2_0_INFO_FILESYSTEM
+    INFO_CLASS = FileSystemInformationClass.FILE_FS_FULL_SIZE_INFORMATION
+
+    def __init__(self):
+        self.fields = OrderedDict([
+            ('total_allocation_units', IntField(size=8)),
```

Comment:
By default an IntField is unsigned but according to the docs for FileFsFullSizeInformation the first 3 fields are a signed integer (Int64). You just need to add `unsigned=False` to the kwargs of IntField for the first 3 fields. The `sectors_per_unit` and `bytes_per_sectore` are documented as unsigned though so those as fine.